Patching Makefiles generated by qmake

IngoMeyer441 opened the pull request:
On OSX Makefiles generated by qmake are patched to remove absolute compiler and linker paths.
See #53

IngoMeyer441 committed:
On OSX Makefiles generated by qmake are patched to remove absolute
compiler and linker paths.

rdnetto commented on config_gen.py:
```diff
@@ -202,6 +202,25 @@ def run(cmd, *args, **kwargs):
         print("$ " + " ".join(cmd))
         subprocess.call(cmd, *args, **kwargs)
 
+    # helper function to set CC and CXX variables correctly on OSX with qmake
```
Please use [PEP8-style](https://www.python.org/dev/peps/pep-0008/#documentation-strings) doc comments.

rdnetto commented on config_gen.py:
```diff
@@ -202,6 +202,25 @@ def run(cmd, *args, **kwargs):
+            "LINK": "clang++"
+        }
+        patched_makefile_lines = []
+        with open(makefile_path, "r") as f:
```
This could be expressed more concisely by applying the `sub()` to the entire file at once, rather than line by line. (The check on `match()` is redundant.)

IngoMeyer441 replied:
I used a combination of `match` and `sub` because so the spaces before `=` are untouched. In the patched Makefile the equal signs are stilled aligned. The `match` is only used for a better style of the patched result and the code block could be replaced by a single `sub` call as you commented before. Or is it possible to use `sub` without changing the style of the original Makefile?

IngoMeyer441 replied:
On the other hand, the patch is applied to a temporary file, so actually style does not matter here. I will change the code.

rdnetto replied:
Add a capture group around the leading whitespace - then you'll be able to prepend it to the generated string.

rdnetto commented on config_gen.py:
```diff
@@ -286,6 +305,8 @@ def run(cmd, *args, **kwargs):
         print("Running qmake in '{}' with Qt {}...".format(build_dir, qt_version))
         run(["qmake"] + configure_opts + [pro_files[0]], env=env_config,
             **proc_opts)
+        if os.uname()[0] == "Darwin":
```
Are you sure that this is needed only for OSX?

IngoMeyer441 replied:
No, I'm not sure. Do you think it's better to remove the if statement?

rdnetto replied:
Well, it works under Linux without it, so it's probably safer to leave it in. I'm not entirely certain why the Makefile has a hardcoded compiler in it, which bothers me, but qmake has a Mac specific configuration/mkspec anyway, so this problem is unlikely to affect users of other platforms.

I think we should be providing a message to the user though, to make them aware of this step - something like:

```
print("Patching generated makefile...")
```

should suffice.

IngoMeyer441 committed:
Altered qmake patch function:
    - PEP8 style doc string
    - Simplified patching (uses only re.sub)
    - Notification is printed when the patching function is invoked
